### PR TITLE
LibJS: Small fixes for Array.prototype.concat

### PR DESCRIPTION
**Pull request by davidot**
Spotted a small typo in concat and then found more issues.
Let me know if I should split changes like this up.

Fixes ~5 test262 tests.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp [hunk 1]
```diff
@@ -551,9 +552,12 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::concat)
                     return;
                 if (k_exists) {
                     auto k_value = obj.get(k).value_or(js_undefined());
-                    if (vm.exception())
+                    if (vm.exception()) {
                         return;
+                    }
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp:557]: Please don't add curly braces to a single-line if body.

File: Userland/Libraries/LibJS/Runtime/Object.cpp [hunk 2]
```diff
@@ -271,8 +271,9 @@ Value Object::get_own_property(const PropertyName& property_name, Value receiver
     if (allow_side_effects == AllowSideEffects::Yes) {
         if (value_here.is_accessor())
             return value_here.as_accessor().call_getter(receiver);
-        if (value_here.is_native_property())
-            return call_native_property_getter(value_here.as_native_property(), receiver);
+        if (value_here.is_native_property()) {
+            return call_native_property_getter(value_here.as_native_property(), this);
+        }
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Object.cpp:276]: Please don't add curly braces to a single-line if body.

File: Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp (hunk 1 above)
Comment (davidot) [reply Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp:557]: Whoops left over from debugging.
Removed them

**Comment by IdanHo** [comment]
Are you sure about the last commit?
```js
const o = {};
Object.freeze(o);
o.foo = "bar";
```
doesnt throw on v8 or webkit for me :thinking:

**Comment by IdanHo** [comment]
Oh i see, youre basing this on `CreateDataPropertyOrThrow`? Not all calls to put_own_property are supposed to go through CreateDataPropertyOrThrow, so throwing inside of it regardless isnt correct behaviour.

**Comment by davidot** [comment]
> Oh i see, youre basing this on `CreateDataPropertyOrThrow`? Not all calls to put_own_property are supposed to go through CreateDataPropertyOrThrow, so throwing inside of it regardless isnt correct behaviour.

Ah that is what I was afraid of. I will have to take a closer look then.

**Comment by davidot** [comment]
I removed the last commit since the throwing behavior is more complicated then it seemed at first glance.
This might be able to be solved with an actual `CreateDataPropertyOrThrow` method because now it is unclear where it should throw.

**Comment by davidot** [comment]
Moving these changes to #8191

**Pull request closed without merging**